src/utils.rs

```diff
@@ -331,13 +371,47 @@ fn internal_exec_cmd(cmd: &str, args: &[&str], time_limit: Duration) -> Option<C
     }
 }
 
+// Bespoke code, inspired by the parallel-stream crate.
+// TODO: replace this with parallel-stream (or similar) when possible.
+// At the time of writing parallel-stream 2.1.2 only works with an old
+// version of async_std, and is missing several features.
+pub async fn parallel_map<I, M, T, Fut, U, C>(items: I, map: M, mut collector: C)
+where
+    I: IntoIterator<Item = T>,
+    M: Fn(T) -> Fut,
+    Fut: Future<Output = U> + Send + 'static,
+    U: Send + 'static,
+    T: Send + 'static,
+    C: FnMut(U),
+{
+    let handles = items
+        .into_iter()
+        .map(|x| {
+            let fut = map(x);
+            async_std::task::spawn(async move { fut.await })
+        })
+        .collect::<Vec<_>>();
+
+    for handle in handles {
+        collector(handle.await);
+    }
+}
+
 #[cfg(test)]
 mod tests {
+    use async_std::task::block_on;
+
     use super::*;
+    fn internal_exec_cmd(cmd: &str, args: &[&str]) -> Option<CommandOutput> {
```

Comment:
I think it would be better to make the tests use [`async_std::test`](https://docs.rs/async-std/1.9.0/async_std/attr.test.html) here.